packages/zoe/src/typeGuards.js

```diff
@@ -1,18 +1,61 @@
-import { AmountShape } from '@agoric/ertp';
+import {
+  AmountShape,
+  AssetKindShape,
+  DisplayInfoShape,
+  IssuerShape,
+  IssuerKitShape,
+  BrandShape,
+  PaymentShape,
+} from '@agoric/ertp';
 import { M } from '@agoric/store';
 import { TimestampValueShape } from '@agoric/swingset-vat/src/vats/timer/typeGuards.js';
+import { SubscriberShape } from '@agoric/notifier';
+
+// keywords have an initial cap
+export const KeywordShape = M.string();
 
 export const InvitationHandleShape = M.remotable('InvitationHandle');
 export const InvitationShape = M.remotable('Invitation');
 export const InstanceHandleShape = M.remotable('InstanceHandle');
 export const InstallationShape = M.remotable('Installation');
 export const SeatShape = M.remotable('Seat');
 
-export const AmountKeywordRecordShape = M.recordOf(M.string(), AmountShape);
+export const AmountKeywordRecordShape = M.recordOf(KeywordShape, AmountShape);
 export const AmountPatternKeywordRecordShape = M.recordOf(
-  M.string(),
+  KeywordShape,
   M.pattern(),
 );
+export const PaymentKeywordRecordShape = M.recordOf(KeywordShape, PaymentShape);
+export const PaymentPKeywordRecordShape = M.recordOf(
+  KeywordShape,
+  M.eref(PaymentShape),
+);
+export const IssuerKeywordRecordShape = M.recordOf(KeywordShape, IssuerShape);
+export const IssuerPKeywordRecordShape = M.recordOf(
+  KeywordShape,
+  M.eref(IssuerShape),
+);
+export const BrandKeywordRecordShape = M.recordOf(KeywordShape, BrandShape);
+
+export const IssuerRecordShape = M.split(
```

Comment:
```suggestion
export const IssuerRecordShape = M.splitRecord(
```

We're deprecating two argument `M.split` and `M.partial` in favor of the three argument `M.splitArray` and `M.splitRecord`.

I am guessing you wanted to make `displayInfo` optional. With `M.split` I don't think this pattern succeeds at that. With `M.splitRecord` it does.

Reply:
Looks done. Resolving.